src/acl_auto_configure.cpp

```diff
@@ -724,9 +724,9 @@ static bool read_accel_defs(const std::string &config_str,
     accel[i].mem.next = reinterpret_cast<void *>(0x00020000);
 
     int total_fields_kernel = 0;
-    if (result) {
-      result = read_int_counters(config_str, curr_pos, total_fields_kernel,
-                                 counters);
+    if (!result || !read_int_counters(config_str, curr_pos, total_fields_kernel,
+                                      counters)) {
+      total_fields_kernel = 0;
     }
```

Comment:
I feel like that would kind of change the logic of it. In the original code, if for example `result == false` before `read_int_counters`, then since it is inside of an if-statement, it would skip `read_int_counters` and push `0` when `counters.emplace_back(total_fields_kernel)`. Then the value would be reset at `read_string_counters`, where `result` could end up as `true`. But if I change them both to `result = result && read....`, then if `result == false`, it would stay false because of the `&&` operator.

Reply:
After a discussion with @zibaiwan, we concluded that `total_fields_kernel` is an essential part of the auto-discovery string since the[ auto-discovery string decoder will read this `total_fields_kernel` to determine the number of fields to parse](https://wiki.ith.intel.com/display/HLD/Auto-discovery+String+Encoding+and+Decoding). Therefore, `result` from `read_int_counters` is non-trivial and cannot be ignored.